source.go

```diff
@@ -312,6 +312,10 @@ func generateSourceFromImage(st llb.State, cmd *Command, sOpts SourceOpts, subPa
 		} else {
 			out = cmdSt.AddMount(subPath, out)
```

Comment:
It seems like the behavior we want is to run each command on the rootfs, and then extract the path afterwards?

Reply:
Sorry, I misread your comment.
In the case that you mention, we'd always be mounting over it regardless.
We'd have to extract that base state out first, then use that instead of scratch.
I think that's a different case than is being fixed here.

Reply:
No problem! I've opened a different issue for this other case, otherwise the fix for this specific bug looks good to me! https://github.com/Azure/dalec/issues/379